flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -181,10 +236,10 @@ public void shutdown(Time timeout) {
 				typeParameters.toArray(new Class<?>[typeParameters.size()]));
 		}
 
-		return submitRequest(targetAddress, targetPort, httpRequest, responseType);
+		return submitRequest(targetAddress, targetPort, finalRequest, requestProcessor, responseType);
 	}
 
-	private <P extends ResponseBody> CompletableFuture<P> submitRequest(String targetAddress, int targetPort, FullHttpRequest httpRequest, JavaType responseType) {
+	private <P extends ResponseBody, T> CompletableFuture<P> submitRequest(String targetAddress, int targetPort, T request, RequestProcessor<T> processor, JavaType responseType) {
```

Comment:
I think we don't have to split the request from the request write logic by splitting `RequestProcessor` and `request` up.